logsmanagement/db_api.c

```diff
@@ -0,0 +1,1096 @@
+    } 
+	// else debug(D_LOGS_MANAG, "Table %s created successfully", MAIN_COLLECTIONS_TABLE);
+    
+    sqlite3_stmt *stmt_search_if_log_source_exists;
+    rc = sqlite3_prepare_v2(main_db,
+                            "SELECT COUNT(*), Id, DB_Dir FROM " MAIN_COLLECTIONS_TABLE
+                            " WHERE Log_Source_Path = ? ;",
+                            -1, &stmt_search_if_log_source_exists, NULL);
+    if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+    
+    sqlite3_stmt *stmt_insert_log_collection_metadata;
+    rc = sqlite3_prepare_v2(main_db,
+                            "INSERT INTO " MAIN_COLLECTIONS_TABLE
+                            " (Machine_GUID, Log_Source_Path, Type, DB_Dir) VALUES (?,?,?,?) ;",
+                            -1, &stmt_insert_log_collection_metadata, NULL);
+    if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+    
+    for (int i = 0; i < p_file_infos_arr->count; i++) {
+
+		/* Initialise DB mutex and acquire lock */
+		p_file_infos_arr->data[i]->db_mut = mallocz(sizeof(uv_mutex_t));
+		rc = uv_mutex_init(p_file_infos_arr->data[i]->db_mut);
+	    if (unlikely(rc)) fatal_libuv_err(rc, __LINE__);
+		db_set_lock(p_file_infos_arr->data[i]->db_mut);
+
+        rc = sqlite3_bind_text(stmt_search_if_log_source_exists, 1, p_file_infos_arr->data[i]->filename, -1, NULL);
+        if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+        rc = sqlite3_step(stmt_search_if_log_source_exists);
+        /* COUNT(*) query should always return SQLITE_ROW */
+        if (unlikely(rc != SQLITE_ROW)) fatal_sqlite3_err(rc, __LINE__);
+        
+        int log_source_occurences = sqlite3_column_int(stmt_search_if_log_source_exists, 0);
+        // debug(D_LOGS_MANAG, "DB file occurences of %s: %d", p_file_infos_arr->data[i]->filename, log_source_occurences);
+        switch (log_source_occurences) {
+            case 0:  /* Log collection metadata not found in main DB - create a new record */
+				
+				/* Bind machine GUID */
+                rc = sqlite3_bind_text(stmt_insert_log_collection_metadata, 1, localhost->machine_guid, -1, NULL);
+                if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+
+				/* Bind log source path */
+                rc = sqlite3_bind_text(stmt_insert_log_collection_metadata, 2, p_file_infos_arr->data[i]->filename, -1, NULL);
+                if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+				
+				/* Bind log type */
+                rc = sqlite3_bind_int(stmt_insert_log_collection_metadata, 3, p_file_infos_arr->data[i]->log_type);
+                if (unlikely(rc != SQLITE_OK)) fatal_sqlite3_err(rc, __LINE__);
+                
+                /* Create directory of collection of logs for the particular 
+				 * log source (in the form of a UUID) and bind it. */
+				uuid_t uuid;
+				uuid_generate(uuid);
+				char uuid_str[GUID_LEN + 1];      // ex. "1b4e28ba-2fa1-11d2-883f-0016d3cca427" + "\0"
```

Comment:
I suggest to use [UUID_STR_LEN](https://github.com/netdata/netdata/blob/fced0fcbefdb3efc905ff1b3660fa09c5c9e6077/libnetdata/libnetdata.h#L106) instead `GUID_LEN`.

Reply:
What's the difference in terms of use between the two? We make sure both are defined as I can see.

Reply:
My suggestion has two goals:

1. To help our friend compiler.
2. You have a variable named uuid, and the definition that I suggested was made for `UUID`, so I think this will simplify a developer reading the code.